src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -114,5 +119,110 @@ private void filterFeedbackSessions(String courseId, List<FeedbackSessionAttribu
+            String key = isCurrentInstructorGiver
+                       ? InstructorStudentRecordsPageData.COMMENT_GIVER_NAME_THAT_COMES_FIRST
+                       : comment.giverEmail;
+
+            List<CommentAttributes> commentList = giverEmailToCommentsMap.get(key);
+            if (commentList == null) {
+                commentList = new ArrayList<CommentAttributes>();
+                giverEmailToCommentsMap.put(key, commentList);
+            }
+            updateCommentList(comment, isCurrentInstructorGiver, commentList, instructor, courseId);
+        }
+        return giverEmailToCommentsMap;
+    }
+    
+    /**
+     * Maps emails of instructors giving the comments to their names.
+     * @param courseId
+     * @param giverEmails
+     * @return A map with instructor email => instructor name mappings.
+     */
+    private TreeMap<String, String> mapGiverNameToGiverEmail(String courseId, Set<String> giverEmails) {
+        TreeMap<String, String> giverEmailToGiverNameMap = new TreeMap<String, String>();
+        giverEmailToGiverNameMap.put(InstructorStudentRecordsPageData.COMMENT_GIVER_NAME_THAT_COMES_FIRST,
+                                     Const.DISPLAYED_NAME_FOR_SELF_IN_COMMENTS);
+        
+        // magic string from CommentsLogic
+        giverEmailToGiverNameMap.put("Anonymous", "Anonymous");
+        for (String giverEmail : giverEmails) {
+            if (!giverEmailToGiverNameMap.containsKey(giverEmail)) {
+                InstructorAttributes giverInstructor = logic.getInstructorForEmail(courseId, giverEmail);
+                Assumption.assertNotNull(giverInstructor);
+                giverEmailToGiverNameMap.put(giverEmail, giverInstructor.displayedName + " " + giverInstructor.name);
+            }
+        }
+        return giverEmailToGiverNameMap;
+    }
+    
+    /**
+     * Adds {@code comment} to {@code commentList} if the instructor is able to view the comment.
+     * @param comment
+     * @param isCurrentInstructorGiver
+     * @param commentList
+     * @param instructor
+     * @param courseId
+     */
+    private void updateCommentList(CommentAttributes comment, boolean isCurrentInstructorGiver,
+                                   List<CommentAttributes> commentList, InstructorAttributes instructor,
+                                   String courseId) {
+        boolean canViewComment = isCurrentInstructorGiver
+                                 || isInstructorAllowedForPrivilegeOnComment(comment, instructor, courseId,
+                                            Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_COMMENT_IN_SECTIONS);
+        if (canViewComment) {
+            commentList.add(comment);
+        }
+    }
+    
+    private boolean isInstructorAllowedForPrivilegeOnComment(CommentAttributes comment, InstructorAttributes instructor,
+                                                             String courseId, String privilegeName) {
```

Comment:
Seems to be possible to shift this logic over to `CommentsLogic` instead, code is minimised since several of the checks here are covered at `CommentsDb` level (one level below) already.

Now exploring possibilities of covering cases where the recipient of the comment is a team/section, so that the newly introduced method behaves as expected for those cases (although such functionality is not required for this PR)